picker/picker.go

```diff
@@ -261,7 +263,7 @@ func (p *Picker) PickAddr() (string, error) {
 	p.r.ObserveIfExists(peer, -1) // downweight the current addr
 
 	var err error
-	peer, err = p.r.Select("")
+	peer, err = p.r.Select()
 	if err != nil {
```

Comment:
@stevvooe ci fails because `Select("")` will exclude peer whose `NodeID` is empty. 
Is this a bug, or we filter out peers who don't have nodeID deliberately？